commit 7a6452a63c344ca17baa2c27f2b26e6dcee84b08 (refs/changes/17/717/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-10-17T16:59:05-07:00 (3 years ago)
    
    PUBAPI-1334 CloudAPI should use cueball for talking to FWAPI
    PUBAPI-1335 fwapi left in the cold with no logger

diff --git a/lib/app.js b/lib/app.js
index e4d7970..06b9ab4 100644
--- a/lib/app.js
+++ b/lib/app.js
@@ -187,6 +187,7 @@ function createClients(options, callback) {
     assert.ok(options.vmapi);
     assert.ok(options.napi);
     assert.ok(options.imgapi);
+    assert.ok(options.fwapi);
     assert.ok(options.papi);
     assert.ok(options.ufds);
     assert.ok(options.ufds_master);
@@ -203,6 +204,9 @@ function createClients(options, callback) {
     options.napi.log   = options.log.child({ component: 'napi' });
     options.imgapi.log = options.log.child({ component: 'imgapi' });
     options.papi.log   = options.log.child({ component: 'papi' });
+    options.fwapi.log  = options.log.child({ component: 'fwapi' });
+
+    options.fwapi.agent = agent;
 
     if (options.mahi) {
         options.mahi.log = options.log.child({ component: 'mahi' });
diff --git a/sapi_manifests/cloudapi/template b/sapi_manifests/cloudapi/template
index e23dead..516ca87 100644
--- a/sapi_manifests/cloudapi/template
+++ b/sapi_manifests/cloudapi/template
@@ -66,7 +66,8 @@
     "cueballHttpAgent": {
         "resolvers": ["{{{BINDER_SERVICE}}}"],
         "initialDomains": [
-            "{{{mahi_domain}}}"
+            "{{{mahi_domain}}}",
+            "{{{fwapi_domain}}}"
         ],
         "spares": 4,
         "maximum": 10,
